### PR TITLE
test(api): refactor e2e label and annotation test case

### PR DESCRIPTION
**Pull request by hardcoretime**
## Description

- adapt existing label and annotation test case to updated API
- fix typo and improve test cases contexts
## Why do we need it, and what problem does it solve?

It adapts `Label and Annotation` test case to updated Virtualization API and this test case does not be skipped in E2E workflow. 
## What is the expected result?

`Label and Annotation` will be checked again in E2E workflow.
## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: tests/e2e/vm_label_annotation_test.go [hunk 1]
```diff
@@ -18,263 +18,295 @@ package e2e
...
-		return cmdResult
+func RemoveLabel(resource kc.Resource, labels map[string]string, names ...string) error {
+	formattedLabels := make([]string, 0, len(labels))
+	for k := range labels {
+		formattedLabels = append(formattedLabels, fmt.Sprintf("%s-", k))
+	}
+	rawResources := strings.Join(names, " ")
+	rawLabels := strings.Join(formattedLabels, "")
+	subCmd := fmt.Sprintf("label %s %s --namespace %s %s", resource, rawResources, conf.Namespace, rawLabels)
+	res := kubectl.RawCommand(subCmd, kc.MediumTimeout)
```
Comment (eofff) [inline tests/e2e/vm_label_annotation_test.go:55]: Why you import with alias kc, but use full form?

**Comment by diafour** [comment]
> Label and Annotation test case will be added to E2E tests workflow.

Is it "Label and Annotation test case is now a part of E2E tests workflow again."

And, shouldn't we unlock this test for `task run` also?

File: tests/e2e/testdata/vm-label-annotation/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: testcases
+namePrefix: pr-number-or-commit-hash-
+resources:
+  - ns.yaml
+  - vi
+  - overlays/vm-labal-annotation
```
Comment (diafour) [inline tests/e2e/testdata/vm-label-annotation/kustomization.yaml:8]: Typo in the directory name: vm-label-annotation

File: tests/e2e/vm_label_annotation_test.go [hunk 3]
```diff
@@ -18,263 +18,295 @@ package e2e
...
-			ApplyFromFile(vmManifest)
-			WaitVmStatus(vm.Name, VMStatusRunning)
-		})
-
-		AfterAll(func() {
-			By("Delete manifest")
-			kubectl.Delete(kc.DeleteOptions{
-				Filename:       []string{vmManifest},
-				FilenameOption: kc.Filename,
+	Context("When resources are applied:", func() {
```
Comment (diafour) [inline tests/e2e/vm_label_annotation_test.go:105]: ":" is not necessary in context description

File: tests/e2e/vm_label_annotation_test.go [hunk 4]
```diff
@@ -18,263 +18,295 @@ package e2e
...
-		Describe(fmt.Sprintf("Add annotation %s=%s", annotationName, annotationValue), func() {
-			It("Annotated", func() {
-				subCMD := fmt.Sprintf("-n %s annotate vm %s %s=%s", conf.Namespace, vm.Name, annotationName,
-					annotationValue)
-				res := kubectl.RawCommand(subCMD, ShortWaitDuration)
-				Expect(res.Error()).NotTo(HaveOccurred())
-			})
-		})
+	Describe("Annotating", func() {
+		Context(fmt.Sprintf("When virtual machines in %s phase:", PhaseRunning), func() {
```
Comment (diafour) [inline tests/e2e/vm_label_annotation_test.go:225]: There is only one Context in this Describe. It seems more convenient to merge them into single Context:
```
Context(fmt.Sprintf("Annotate VirtualMachines in %s phase", PhaseRunning), func() ...
```

File: tests/e2e/vm_label_annotation_test.go [hunk 5]
```diff
@@ -18,263 +18,295 @@ package e2e
...
-				Expect(res.Error()).NotTo(HaveOccurred(), "failed to get KVVMI %s.\n%s", vm.Name, res.StdErr())
-				Expect(res.StdOut()).To(BeEmpty())
-			})
-			It("POD virtlauncher", func() {
-				pod := getPodName(vm.Name)
-				res := getRecourseAnnotation(kc.ResourcePod, pod)
-				Expect(res.Error()).NotTo(HaveOccurred(), "failed to get pod %s.\n%s", vm.Name, res.StdErr())
-				Expect(res.StdOut()).To(BeEmpty())
+	Context("When test is complited:", func() {
+		It("tries to delete used resources", func() {
```
Comment (diafour) [inline tests/e2e/vm_label_annotation_test.go:301]:
```suggestion
	Context("When test is completed", func() {
		It("should delete all created resources", func() {
```

Comment (diafour) [inline tests/e2e/vm_label_annotation_test.go:301]: Also,  cleanup operation seems strange to be a part of the test suite. Shouldn't it belongs to the test setup actions, something like AfterAll?

**Review by eofff — APPROVED** [review]

File: tests/e2e/vm_label_annotation_test.go (hunk 5 above)
Comment (hardcoretime) [reply tests/e2e/vm_label_annotation_test.go]: I put it to wrapper function with description. We need this context to check that all resources in test cases can be deleted correctly.

**Comment by hardcoretime** [comment]
> > Label and Annotation test case will be added to E2E tests workflow.
> 
> Is it "Label and Annotation test case is now a part of E2E tests workflow again."?
> 
> And, shouldn't we unlock this test for `task run` also?

Yes, we unlock this test in this PR.

**Review by eofff — APPROVED** [review]

**Pull request merged**